version.cmake

```diff
@@ -101,29 +101,47 @@ MACRO(VERSION_COMPUTE)
 
     # Check if git describe worked and store the returned version number.
     IF(GIT_DESCRIBE_RESULT)
-      MESSAGE(
-	"Warning: failed to compute the version number,"
-	" 'git describe' failed:\n"
-	"\t" ${GIT_DESCRIBE_ERROR})
+      MESSAGE(AUTHOR_WARNING
+        "Warning: failed to compute the version number,"
+        " 'git describe' failed:\n"
+        "\t" ${GIT_DESCRIBE_ERROR})
       SET(PROJECT_VERSION UNKNOWN)
     ELSE()
       # Get rid of the tag prefix to generate the final version.
       STRING(REGEX REPLACE "^v" "" PROJECT_VERSION "${GIT_DESCRIBE_OUTPUT}")
       IF(NOT PROJECT_VERSION)
-	MESSAGE(
-	  "Warning: failed to compute the version number,"
-	  "'git describe' returned an empty string.")
-	SET(PROJECT_VERSION UNKNOWN)
+	      MESSAGE(AUTHOR_WARNING
+	        "Warning: failed to compute the version number,"
+          "'git describe' returned an empty string.")
+        SET(PROJECT_VERSION UNKNOWN)
       ENDIF()
 
       # If there is a dash in the version number, it is an unstable release,
       # otherwise it is a stable release.
       # I.e. 1.0, 2, 0.1.3 are stable but 0.2.4-1-dg43 is unstable.
       STRING(REGEX MATCH "-" PROJECT_STABLE "${PROJECT_VERSION}")
       IF(NOT PROJECT_STABLE STREQUAL -)
-	SET(PROJECT_STABLE True)
+        SET(PROJECT_STABLE True)
       ELSE()
-	SET(PROJECT_STABLE False)
+        SET(PROJECT_STABLE False)
+      ENDIF()
+    ENDIF()
+
+    # Compute the major, minor and patch version of the project
+    IF(NOT DEFINED PROJECT_VERSION_MAJOR AND
+       NOT DEFINED PROJECT_VERSION_MINOR AND
+       NOT DEFINED PROJECT_VERSION_PATCH)
+      IF(PROJECT_VERSION MATCHES UNKNOWN)
+        SET(PROJECT_VERSION_MAJOR UNKNOWN)
+        SET(PROJECT_VERSION_MINOR UNKNOWN)
+        SET(PROJECT_VERSION_PATCH UNKNOWN)
+      ELSE()
+        # Extract the version from PROJECT_VERSION
+        string(REPLACE "." ";" _PROJECT_VERSION_LIST "${PROJECT_VERSION}")
+        string(REPLACE "-" ";" _PROJECT_VERSION_LIST "${_PROJECT_VERSION_LIST}")
+        list(GET _PROJECT_VERSION_LIST 0 PROJECT_VERSION_MAJOR)
+        list(GET _PROJECT_VERSION_LIST 1 PROJECT_VERSION_MINOR)
+        list(GET _PROJECT_VERSION_LIST 2 PROJECT_VERSION_PATCH)
```

Comment:
What happens here if the version does not have a patch number (e.g. `1.0`)?

Reply:
I just push a fix for that.